core/src/seeding/seeding_algorithm.cpp

```diff
@@ -12,60 +12,24 @@
 
 // System include(s).
 #include <cmath>
-
-namespace {
-
-/// Helper function that would produce a default seed-finder configuration
-traccc::seedfinder_config default_seedfinder_config() {
-
-    traccc::seedfinder_config config;
-    traccc::seedfinder_config config_copy = config.toInternalUnits();
-    config.highland = 13.6 * std::sqrt(config_copy.radLengthPerSeed) *
-                      (1 + 0.038 * std::log(config_copy.radLengthPerSeed));
-    float maxScatteringAngle = config.highland / config_copy.minPt;
-    config.maxScatteringAngle2 = maxScatteringAngle * maxScatteringAngle;
-    // helix radius in homogeneous magnetic field. Units are Kilotesla, MeV
-    // and millimeter
-    config.pTPerHelixRadius = 300. * config_copy.bFieldInZ;
-    config.minHelixDiameter2 =
-        std::pow(config_copy.minPt * 2 / config.pTPerHelixRadius, 2);
-    config.pT2perRadius =
-        std::pow(config.highland / config.pTPerHelixRadius, 2);
-    return config;
-}
-
-/// Helper function that would produce a default spacepoint grid configuration
-traccc::spacepoint_grid_config default_spacepoint_grid_config() {
-
-    traccc::seedfinder_config config = default_seedfinder_config();
-    traccc::spacepoint_grid_config grid_config;
-    grid_config.bFieldInZ = config.bFieldInZ;
-    grid_config.minPt = config.minPt;
-    grid_config.rMax = config.rMax;
-    grid_config.zMax = config.zMax;
-    grid_config.zMin = config.zMin;
-    grid_config.deltaRMax = config.deltaRMax;
-    grid_config.cotThetaMax = config.cotThetaMax;
-    grid_config.impactMax = config.impactMax;
-    grid_config.phiMax = config.phiMax;
-    grid_config.phiMin = config.phiMin;
-    grid_config.phiBinDeflectionCoverage = config.phiBinDeflectionCoverage;
-    return grid_config;
-}
-
-}  // namespace
+#include <iostream>
 
 namespace traccc {
 
-seeding_algorithm::seeding_algorithm(vecmem::memory_resource& mr)
-    : m_spacepoint_binning(default_seedfinder_config(),
-                           default_spacepoint_grid_config(), mr),
-      m_seed_finding(default_seedfinder_config(), seedfilter_config()) {}
+seeding_algorithm::seeding_algorithm(vecmem::memory_resource& mr) : m_mr(mr) {
+    m_finder_config.setup();
+    m_grid_config.setup(m_finder_config);
+}
 
 seeding_algorithm::output_type seeding_algorithm::operator()(
     const spacepoint_collection_types::host& spacepoints) const {
 
-    return m_seed_finding(spacepoints, m_spacepoint_binning(spacepoints));
+    /// Sub-algorithm performing the spacepoint binning
+    spacepoint_binning binning_alg(m_finder_config, m_grid_config, m_mr);
+    /// Sub-algorithm performing the seed finding
+    seed_finding finding_alg(m_finder_config, m_filter_config);
+
+    return finding_alg(spacepoints, binning_alg(spacepoints));
```

Comment:
Why? Why would you want to (re-)create the `traccc::spacepoint_binning` and `traccc::seed_finding` algorithms on every call of the operator? :confused:

Let's keep the setup exactly like we had before. The only thing that I can see being useful is to extend the constructor of `traccc::seeding_algorithm` to receive non-standard configurations. Since right now there is no way to specify a non-standard configuration. :frowning:

But we should keep the `m_spacepoint_binning` and `m_seed_finding` members as they are. Let's not get rid of them!

Reply:
I reverted the change but seeding will take config parameters as inputs instead. Please tell me if you are OK with that.
This will require some changes in `full_chain_algorithm` constructor and etc.